modules/api/php/endpoints/candidate/visit/visit_0_0_4_dev.class.inc

```diff
@@ -0,0 +1,155 @@
+<?php declare(strict_types=1);
+/**
+ * This implements the Visit endpoint class under Candidates/$candid
+ *
+ * PHP Version 7
+ *
+ * @category API
+ * @package  Loris
+ * @author   Xavier Lecours Boucher <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://github.com/aces/Loris
+ */
+namespace LORIS\api\Endpoints\Candidate\Visit;
+
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+
+/**
+ * A class for handling the /candidates/$candidate/$visit_label endpoint.
+ *
+ * @category API
+ * @package  Loris
+ * @author   Xavier Lecours Boucher <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://github.com/aces/Loris
+ */
+class Visit_0_0_4_Dev extends Visit
+{
+    /**
+     * Return which methods are supported by this endpoint.
+     *
+     * @return array supported HTTP methods
+     */
+    protected function allowedMethods() : array
+    {
+        return [
+            'GET',
+            'PATCH',
+            'PUT',
+        ];
+    }
+
+    /**
+     * Versions of the LORIS API which are supported by this
+     * endpoint.
+     *
+     * @return array a list of supported API versions.
+     */
+    protected function supportedVersions() : array
+    {
+        return [
+            'v0.0.4-dev'
+        ];
+    }
+
+    /**
+     * Handles a PATCH request.
+     * For now, only support a NextStageDate argument to start next stage.
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    protected function handlePATCH(
+        ServerRequestInterface $request
+    ): ResponseInterface {
+        $data          = json_decode((string) $request->getBody(), true);
+        $nextStageDate = $data['NextStageDate'] ?? null;
+
+        if ($this->_visit === null) {
+            return new \LORIS\Http\Response\JSON\NotFound('Visit not found');
+        }
+
+        $response = $this->validatePUT($request);
+        if ($response !== null) {
+            return $response;
+        }
+
+        if (!isset($data['Visit'])
+            || $data['Visit'] !== $this->_visit->getVisitLabel()
+        ) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'The Visit field does not match this visit'
+            );
+        }
+
+        if (!$nextStageDate) {
+            return new \LORIS\Http\Response\JSON\BadRequest(
+                'There is no stage date specified'
+            );
+        }
+
+        $date_format = 'Y-m-d';
+        $date        = \DateTime::createFromFormat($date_format, $nextStageDate);
+
+        if (!($date && $date->format($date_format) === $nextStageDate)) {
```

Comment:
Is there a case when `$date->format($date_format) === $nextStageDate` would not be true?

Reply:
> This should also modify `modules/api/static/schema.yml`. I can push a contribution if you would like me to.

I am not modifying v0.0.3 so do we need to change anything?